### PR TITLE
Implement linabo-build-system 

### PR DESCRIPTION
**Pull request by RedLeader962**
# Description

### Summary:

Add `norlab-build-system` library support. _libnabo_ now have a build infrastructure agnostic build system which can be run on continuous integration infrastructure such as github-action or TeamCity Continuous Integration server. It can also be run localy to improve developement workflow. It support multi-architecture and multi-os version build.
There is also new installer script for ubuntu

### Changes and type of changes (quick overview):

- build: add git submodule N2ST and NBS [NMO-464]
- feat: integrate norlab-build-system library and implement linabo-build-system [NMO-405]
- feat: implement docker image (refactor libpointmatcher-dependencies images) [NMO-406]
- ci: setup teamcity libnabo project [NMO-407]
- ci: setup norlab/libnabo repository branching scheme [NMO-408]
- docs: write contributiong section in readme or add contributing.md [NMO-466]
- build: Added norlab-build-system submodule to repository [NMO-405]
- chore: added a code owner designation file, a pull request template and a conventional commit
  reference file

---

# Checklist:

### Code related

- [x] I have made corresponding changes to the documentation 
      (i.e.: function, class, script header, README.md)
- [x] I have commented hard-to-understand code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All tests pass locally with my changes 
      (Check [contributing/contributing_instructions.md](https://github.com/norlab-ulaval/libnabo/tree/develop/contributing/contributing_instructions.md) for local testing procedure using _libnabo-build-system_)

### PR creation related

- [x] My pull request `base ref` branch is set to the `develop` branch 
     (the _build-system_ won't be triggered otherwise)
- [x] My pull request branch is up-to-date with the `develop` branch 
     (the _build-system_ will reject it otherwise)

### PR description related

- [x] I have included a quick summary of the changes
- [x] I have indicated the related issue's id with `# <issue-id>` if changes are of type `fix`
- [x] I have included a high-level list of changes and their corresponding types
      (See [contributing/commit_msg_reference.md](https://github.com/norlab-ulaval/libnabo/blob/develop/contributing/commit_msg_reference.md) for details)

---

### CODE REVIEW

File: build_system/tests/tests_bats/test_dotenv_libpointmatcher.bats
Comment (boxanm) [inline build_system/tests/tests_bats/test_dotenv_libpointmatcher.bats:1]: This one is called libpointmatcher

File: libnabo_installer.bash [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+#
+# Install libnabo
+#
+# Usage:
+#   $ bash libnabo_installer.bash [<optional flag>]
+#
+# Arguments:
+#   --install-path </dir/abs/path/>     The directory where to install libnabo (absolute path)
+#                                           (default location defined in the .env)
+#   --repository-version 1.0.7         Install libnabo release tag version (default to master branch latest)
```
Comment (boxanm) [inline libnabo_installer.bash:12]: Can we update the tag to the planned release? (1.1.0)

**Review by boxanm — CHANGES_REQUESTED** [review]
Looking good! 👍  I'd consider running find and replace for any leftover `libpointmatcher` keywords and old version numbers (`1.0.7` was the last release, I believe)

File: libnabo_installer.bash (hunk 1 above)
Comment (RedLeader962) [reply libnabo_installer.bash:12]: @boxanm This is a documentation example on how to use this flag, could be any random number that fit the pattern.

**Comment by RedLeader962** [comment]
@boxanm For in `bats` tests, we can't change references to the latest release tag (1.0.7) as there is test logic that require fetching an actual released version. For bash script, the one mentioned in `libnabo_installer.bash` header documentation is there just as an examples of the proper pattern to pass as argument. The two places where we need to bump the version number (ie `package.xml` and `nabo.h`) will be modified at the final release step.

Thanks for spotting the last sneaky libpointmatcher from the codebase, I've renamed it to `test_dotenv_libnabo.bats`.

**Review by boxanm — DISMISSED** [review]

**Comment by RedLeader962** [comment]
I've also made minor style change in `Dockerfile.libnabo.integration_test` and `docker-compose.libnabo.yaml` so that it be intuitive i.e. change `.*compilation_test.*` to `.*integration_test*`

I'll push the commit tonight

**Review by simonpierredeschenes — APPROVED** [review]

**Pull request merged**